### PR TITLE
Don't trigger discover from json file if it doesn't exist

### PR DESCRIPTION
**Pull request by igaw**
fabrics: check if json config is existing

nvme_read_config will return success even if the file doesn't exist.
This leads to the situation that we always execute
discover_from_json_config_file even though there is no json config
file.

One side effect of this is that when executing 'nvme discover -d
/dev/nvmeXXX' we will always do a discover for the first controller we
find.
 
While at it also resolve duplicate alias  command line options.

Fixes: #1992

### CODE REVIEW

File: fabrics.c [hunk 1]
```diff
@@ -730,7 +739,7 @@ int nvmf_discover(const char *desc, int argc, char **argv, bool connect)
 	if (context)
 		nvme_root_set_application(r, context);
 
-	if (!nvme_read_config(r, config_file))
```
Comment (ikegami-t) [inline fabrics.c:733]: Seems okay to check the errno value as below.
```c
    errno = 0;
    if (!nvme_read_config(r, config_file) && !errno)
```

Comment (igaw) [reply fabrics.c:733]: `nvme_read_config` should actually not touch errno if there is failure. This is actually a bug in libnvme... Good catch!

Comment (ikegami-t) [reply fabrics.c:733]: Yes agreed. Thank you.

**Comment by ikegami-t** [comment]
Looks good. (Note: 1 checkpatch line length warnning error)

**Comment by igaw** [comment]
> Looks good. (Note: 1 checkpatch line length warnning error)

Saw it, but it looks better if we leave the formatting as it is IMO.

**Pull request merged**